x/hub-genesis/keeper/ibc_module.go

```diff
@@ -116,20 +117,100 @@ func (w IBCModule) OnAcknowledgementPacket(
 		return err
 	}
 
+	// validate it's a channel we are expecting
+	portChannel := types.PortAndChannel{Port: packet.SourcePort, Channel: packet.SourceChannel}
+	isPending, err := w.k.IsPendingChannel(ctx, portChannel)
+	if err != nil {
+		return errorsmod.Wrap(err, "check pending channel")
+	}
+	if !isPending {
+		// Not supposed to happen anyway
+		w.logger(ctx).Error("acknowledgement failed for unknown channel", "packet", packet, "ack", ack)
+		return errors.New("acknowledgement failed for unknown channel")
+	}
+
 	if !ack.Success() {
 		w.logger(ctx).Error("acknowledgement failed for genesis transfer", "packet", packet, "ack", ack)
-		return errors.New("acknowledgement failed for genesis transfer")
+
+		// Mark the channel for retry
+		portChan := types.PortAndChannel{Port: packet.SourcePort, Channel: packet.SourceChannel}
+		if err := w.k.SetPendingChannel(ctx, portChan, types.NeedsRetry); err != nil {
+			return errorsmod.Wrap(err, "set channel retry required")
+		}
+
+		var gbData types.GenesisBridgeData
+		err := json.Unmarshal(packet.Data, &gbData)
+		if err != nil {
+			return errorsmod.Wrap(err, "unmarshal genesis bridge data")
+		}
+
+		if gbData.GenesisTransfer != nil {
+			// unescrow and refund the funds
+			err = w.k.UnescrowGenesisTransferFunds(ctx, packet.SourcePort, packet.SourceChannel, gbData.GenesisInfo.BaseCoinSupply())
+			if err != nil {
+				return errorsmod.Wrap(err, "unescrow genesis transfer funds")
+			}
+		}
+
+		return nil
 	}
 
-	w.k.enableBridge(ctx, state)
+	// If ack success, enable the bridge for the successful channel and clean up pending state
+	if err := w.k.ClearPendingChannels(ctx); err != nil {
+		return errorsmod.Wrap(err, "clear pending channels")
+	}
+	w.k.enableBridge(ctx, state, packet.SourcePort, packet.SourceChannel)
+
 	ctx.EventManager().EmitEvent(sdk.NewEvent(types.EventTypeOutboundTransfersEnabled))
 	w.logger(ctx).Info("genesis bridge phase completed successfully")
 
 	return nil
 }
 
-// enableBridge enables the bridge after successful genesis bridge phase.
-func (k Keeper) enableBridge(ctx sdk.Context, state types.State) {
-	state.OutboundTransfersEnabled = true
-	k.SetState(ctx, state)
+// OnTimeoutPacket handles IBC packet timeouts
+func (w IBCModule) OnTimeoutPacket(
```

Comment:
I think it's redundant.. no need to care about timeouts just ack errors.

Reply:
fixed